src/multilayerqg.jl

```diff
@@ -533,16 +535,50 @@ Compute the inverse Fourier transform of `varh` and store it in `var`.
 """
 invtransform!(var, varh, params::AbstractParams) = ldiv!(var, params.rfftplan, varh)
 
+"""
+    @kernel function pvfromstreamfunction_kernel!(qh, ψh, S, nlayers)
+
+Kernel for GPU acceleration of PV from streamfunction calculation, i.e., obtaining the Fourier
+transform of the PV from the streamfunction `ψh` in each layer using `qh = params.S * ψh`.
+"""
+@kernel function pvfromstreamfunction_kernel!(qh, ψh, S, nlayers)
+  i, j = @index(Global, NTuple)
+
+  @unroll for k = 1:nlayers
+
+      @inbounds qh[i, j, k] = 0
+
+      @unroll for m = 1:nlayers
```

Comment:
The `@unroll` don't do anything unless `nlayers` is known at compile time (this requires using `Val`, but I don't know if it will speed anything up... it might).